chalicelib/action.py

```diff
@@ -81,45 +81,48 @@ def _unsupported_action(self):
 
     def _add_action(self):
 
-        if len(self.params) < 3 or len(self.params) > 4:
-            log.debug(f"params: {self.params}")
+        # validate number of arguments
+        if not self._valid_number_of_args(min_args=2, max_args=3):
+            log.debug(f"args: {self.arguments}")
             return self.send_response(message="Wrong number of args for add command")
 
-        reason = self.params[1]
-        date_string = self.params[2]
-        hours = 8
-
-        if len(self.params) == 4:
-            try:
-                hours = round(float(self.params[3]))
-            except ValueError:
-                return self.send_response(message="Could not parse hours")
+        # assign
+        reason: str = self.arguments[0]
+        input_date: str = self.arguments[1]
+        hours: str = self.arguments[2] if len(self.arguments) == 3 else 8
 
+        # validate reason
         if not self._valid_reason(reason=reason):
             return self.send_response(message=f"Reason {reason} is not valid")
 
-        date = parse_date(date=date_string, format_str=self.format_str)
+        # validate hours
+        try:
+            hours: int = round(float(hours))
+        except ValueError:
+            return self.send_response(message=f"Could not parse hours: {hours}")
 
-        if not date:
+        # validate dates
+        parsed_dates: Dict[str, datetime] = parse_date(
```

Comment:
In other actions we call this variable only `date`. Might be good to be consistent

Reply:
The problem with the name "date" is that it does not convey the state or data type. 

Since we have input date from arguments, parsed dates and datetime objects it makes it hard to understand what "date" means in the context.

Reply:
Sure, so change the others to `parsed_date` then 😄